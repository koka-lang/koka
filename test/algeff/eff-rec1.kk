effect rec cow {
  fun moo() : (() -> <cow,div>  ())
}

val cow = handler {
    moo() -> {
        println("moo")
        resume( {moo()() } )
    }
}

fun main() {
    cow {
        moo()()
    }
}
