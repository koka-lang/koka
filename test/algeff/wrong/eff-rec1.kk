effect cow {
  fun moo() : (() -> cow ())
}

val cow = handler {
    moo() -> {
        println("moo")
        resume( {moo()() } )
    }
}

fun main() {
    cow {
        moo()()
    }
}
