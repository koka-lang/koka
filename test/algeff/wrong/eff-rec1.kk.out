test/algeff/wrong/eff-rec1.kk(1, 1): error: Type .ops-cow is declared as being (co)inductive but it occurs
 recursively in a negative position.
 hint: declare it as a 'type rec' (or 'effect rec)' to allow negative occurrences